opensquirrel/ir.py

```diff
@@ -69,20 +69,40 @@ class Expression(IRNode, ABC):
     pass
 
 
-@dataclass
+@dataclass(init=False)
```

Comment:
Maybe for a different ticket, but it seems like we are converting all dataclasses to normal classes, i.e. why create an __init__ explicitly and not use the __post_init__?

Reply:
Sure, I just tried to write both classes consistently.

Reply:
I figured as much. Stan might have had an idea here for the dataclasses, I don't know. We should have a look at this at some point then (like a general refactor)